Added console warning for `Table` when both `subRows` and `subComponent` exist

smmr-dn opened the pull request:
<!--
Thank you for your contribution to iTwinUI.

If you are only making changes to the docs site using the "Edit page on GitHub" link,
then you can ignore most of this template.
-->

## Changes

<!--
What kind of code changes does this PR include?
Mention anything that could be helpful for reviewers and include screenshots for visual changes.
-->

Added console warning when users pass in `subRows` and `subComponent` for the `Table` at the same time, mentioned in one of the comments in this PR: https://github.com/iTwin/iTwinUI/pull/2208.

## Testing

<!--
How did you test your changes?
If your PR has visual changes, then make sure they are demonstrated in css-workshop and react-workshop, then approve visual test images for both (`pnpm approve:css` and `pnpm approve:react`).

If not applicable, you can write "N/A".
-->

Verified that the warning appears when there are both `subRows` and `subComponent` in the `Table`.

## Docs

<!--
If your PR includes user-facing changes, then update docs in all places (JSDoc, website, stories, etc).
Make sure to include a changeset (`pnpm changeset`).

If not applicable, you can write "N/A".
-->

Added changeset

mayank99 commented on packages/itwinui-react/src/core/Table/Table.tsx:
```diff
@@ -656,6 +656,12 @@ export const Table = <
     }
   }
 
+  if (data.some((item) => (item.subRows as T[]).length > 0) && subComponent) {
```
Can you also add a check for `process.env.NODE_ENV === 'development'`?

smmr-dn replied:
I have added the if condition to check for the dev environment.

mayank99 replied:
it should be made part of the same `if` condition instead of a nested one.

smmr-dn replied:
I have made the change.

mayank99 reviewed (approved):
LGTM. I verified that the warning appears when using both `subRows` and `subComponent`.

![](https://github.com/user-attachments/assets/b427d531-46a0-4856-b615-1dde245aff98)


You should also mention in the PR description how you tested it.

Also, this is a user-facing change, so I think it needs a `patch` changeset.

smmr-dn commented:
> LGTM. I verified that the warning appears when using both `subRows` and `subComponent`.
> 
> ![](https://private-user-images.githubusercontent.com/9084735/374309250-b427d531-46a0-4856-b615-1dde245aff98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzMzI0MjAsIm5iZiI6MTcyODMzMjEyMCwicGF0aCI6Ii85MDg0NzM1LzM3NDMwOTI1MC1iNDI3ZDUzMS00NmEwLTQ4NTYtYjYxNS0xZGRlMjQ1YWZmOTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDdUMjAxNTIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzhlZDcwODZmZTU1M2UwMjI1ZWI4NTkzYzJiZGM5NjkxNzExY2ViYjMzMDRlM2ZiN2FlNWJmMjIxZmY5ZmRkYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.KhRqYyNaXYaoWyWvO0kaqGQfjciTsbbWOzEh4Nd2LVo)
> 
> You should also mention in the PR description how you tested it.
> 
> Also, this is a user-facing change, so I think it needs a `patch` changeset.

I have modified the PR description and added a changeset.

smmr-dn commented on packages/itwinui-react/src/core/Table/Table.tsx:
```diff
@@ -656,6 +656,16 @@ export const Table = <
+  if (
+    process.env.NODE_ENV === 'development' &&
+    data.some((item) => (item.subRows as T[]).length > 0) &&
+    subComponent
```
I have made the change.
